test/e2e/common/node/pods.go

```diff
@@ -1048,7 +1052,12 @@ var _ = SIGDescribe("Pods", func() {
 			}
 			return false, nil
 		})
-		framework.ExpectNoError(err, "failed to see %v event", watch.Deleted)
+		if err != nil {
+			framework.Logf("failed to see %v event: %v", watch.Deleted, err)
+		}
+		_, err = f.ClientSet.CoreV1().Pods(testNamespaceName).Get(context.TODO(), testPodName, metav1.GetOptions{})
```

Comment:
the delete watcher above is incorrect, because it doesn't handle DeletedFinalStateUnknown objects